src/H5Tpkg.h

```diff
@@ -41,9 +41,10 @@
 /* Other public headers needed by this file */
 #include "H5Spublic.h" /* Dataspace functions			*/
 
-/* Macro to ease detecting "complex" datatypes (i.e. those with base types or fields) */
-#define H5T_IS_COMPLEX(t)                                                                                    \
-    ((t) == H5T_COMPOUND || (t) == H5T_ENUM || (t) == H5T_VLEN || (t) == H5T_ARRAY || (t) == H5T_REFERENCE)
+/* Macro to ease detecting composite datatypes (i.e. those with base types or fields) */
+#define H5T_IS_COMPOSITE(t)                                                                                  \
```

Comment:
Renamed this macro from "H5T_IS_COMPLEX" -> "H5T_IS_COMPOSITE" to distinguish from complex number datatypes.